scripts/us_fda/drugs/README.md

```diff
@@ -0,0 +1,108 @@
+### License
+Drugs@FDA is public domain and made available with a Creative Commons CC0 1.0 Universal dedication.
+
+## Scripts
+### create_enums.py
+
+Functions involving creating the mcf for related enums and loading the necessary enum dictionaries can be found in `create_enums.py`.
+
+Running `create_enums.py` from command line will write the file FDADrugsEnumSchema.mcf to the current directory.
+
+Required files for `create_enums.py` are FDADrugsDosageForms.csv and FDADrugsAdminRoutes.csv .
+
+FDADrugsEnumsSchema.mcf contains the schema for the following enums:
+
+* AdministrationRouteEnum
+* DosageFormEnum
+* TherapeuticEquivalenceCodeEnum
+* MarketingStatusEnum
+* ApplicationTypeEnum
+
+### clean_data.py
+
+Functions involving combining original data from different files and writing cleaned data to CleandData.csv can be found in `clean_data.py`.
+
+Required files are Products.txt, Applications.txt, TE.txt and MarketingStatus.txt.
+
+This script loads the dictionaries created in `create_enums.py` in order to map Application-Number + Product-Number to MarketingStatus and TECode enums.
+
+### write_mcf.py
+
+Functions involving parsing the clean data from CleanData.csv and writing the final mcf file for the FDA Drugs Import can be found in `write_mcf.py`.
+
+Required files are CleanData.csv and chemblIDs.out .
+
+This script loads the dictionaries created in `create_enums.py` in order to map Application Types, Dosage Forms, and Administration Routes to enums.
+
+If CleanData.csv is already available, then executing `write_mcf.py` by itself will pull the necessary dictionaries from `create_enums.py` and generate FDADrugsFinal.mcf .
+
+### run_all.py
+
+Executing `run_all.py` from the command line will execute the above scripts, generating FDADrugsDosageForms.mcf, CleanData.csv and FDADrugsFinal.mcf .
+
+
+## AdditionalInformation
+
+Note: `write_mcf.py` takes ~2hrs to run due to the queries to the ChEMBL database using their python API.
+
+chemblIDs.out (used to map drug names to their ChEMBL IDs) originates from chembl_27.0_molecule.ttl found at ftp://ftp.ebi.ac.uk/pub/databases/chembl/ChEMBL-RDF/latest/ 
+
+```
+$ awk '{ if ($0~/^chembl_molecule:/) { molecule=$1; } else if ($0=="") { molecule=""; } if (molecule != "" && $0~/rdfs:label/) { print molecule, $0; }}' chembl_27.0_molecule.ttl > chemblIDs.out    
+$ awk '$3 !~ "CHEMBL"' chemblIDs.out > chembl_ids.out
+```
```

Comment:
We would ideally have a single py or shell-script that downloads all the external files into raw_data/ and does such pre-processing.